indigo_app/js/components/DocumentTOCView.vue

```diff
@@ -239,8 +214,21 @@ export default {
       }
     },
 
+    handleItemTitleClicked (e) {
+      this.selectItem(e.target.item.index, true);
+    },
+
     // select the i-th item in the TOC
     selectItem (i, force) {
+      // Update toc item state
+      const tocItems = this.$refs['la-toc-controller'].querySelectorAll('la-toc-item');
+      for (const tocItem of tocItems) {
+        tocItem.classList.remove('selected');
+        if (tocItem.item.index === i) {
+          tocItem.classList.add('selected');
+        }
+      }
+
       const index = this.selection.get('index');
 
       i = Math.min(this.toc.length - 1, i);
```

Comment:
Does this old logic of setting `.selected` on the TOC entry still apply?

Does the selection get preserved when the TOC is re-rendered after an edit?

Reply:
Maybe it would be  easier to respond to the on item rendered event to add the selection class, and re-render the entire tree when the selection changes?

Reply:
It looks like this works, I'm not sure why :D

Reply:
added listener 
```
this.selection.on('change', this.updateLaToc);
```
so listens for selection change that it will update the web component

Reply:
This seems like only part of the picture to me. `this.selection` is currently how the component communicates the selection to the outside world -- it doesn't actually respond to changes in that (otherwise we'd have had an event handler on that before). So it's a bit of a red herring.

The bigger issue for me is the `.selected` attribute on the TOC entries. That is currently being set but not used. What I was talking about this morning was actually using that to set the `selected` class in `onItemRendered`. If we're not using `.selected` then it should be removed. If we ARE going to drive selections  based on it, then we don`t need to walk over the `la-toc-item` elements and set classes through that mechanism.

Basically: chose one mechanism to declare how we know which item is selected, and use it throughout.

Reply:
ok, I have chosen to go with using the `index` to determine which toc item gets the `selected` CSS class. the `attribute` `selected` has been removed and its associated logic